### PR TITLE
Update ColorControl bounds based on spec fixes

### PR DESCRIPTION
**Pull request by tcarmelveilleux**
### Problem

Minimum color temperature bounds in spec were too wide and included
a value for ColorTempPhysicalMinMireds that could lead to divide-by-zero
in examples (and it did). 

The spec has been fixed (https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10202) and test plan updated to avoid the zero value (https://github.com/CHIP-Specifications/chip-test-plans/pull/4545).

### This PR

- Updates the SDK and test script to make 0 an invalid value for ColorTempPhysicalMinMireds
- Adds internal safety bound checks to prevent things from falling through the cracks even if
  attributes are written to invalid values.
- Updates the XML to make the value 0 forbidden where it should be forbidden.
- Makes all examples have reasonable Min/Max claimed physical values (2000K..6500K --> 154 to 454 mireds) as suggested  by Signify (these values can be made different, but the previous defaults were actively wrong and never correct).
  - Note that the ZAP files from SDK have been widely copied, and the exact sample values have been seen in commercially available products, so fixing the examples if the most important part.

### Testing done

- All cert tests still pass
- Validated that with a value of 0, there is no crash

### CODE REVIEW

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.matter
Comment (tcarmelveilleux) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.matter:9167]: Do not merge (or resolve this comment) until spec PR and test plan PR are fully merged.

**Review by cecille — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_CC_6_5.yaml
Comment (ReneJosefsen) [inline src/app/tests/suites/certification/Test_TC_CC_6_5.yaml:81]: This one also needs to be minValue: 1, right?

**Review by ReneJosefsen — APPROVED** [review]

**Pull request merged**
